services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentControllerITests.kt

```diff
@@ -0,0 +1,150 @@
+package fi.hel.haitaton.hanke.liitteet
+
+import fi.hel.haitaton.hanke.ControllerTest
+import fi.hel.haitaton.hanke.HankeService
+import fi.hel.haitaton.hanke.IntegrationTestConfiguration
+import fi.hel.haitaton.hanke.currentUserId
+import fi.hel.haitaton.hanke.factory.AttachmentFactory
+import fi.hel.haitaton.hanke.factory.DateFactory
+import fi.hel.haitaton.hanke.permissions.PermissionCode
+import fi.hel.haitaton.hanke.permissions.PermissionService
+import io.mockk.*
+import java.util.*
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.context.annotation.Import
+import org.springframework.mock.web.MockMultipartFile
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.ResultActions
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+private const val username = "username"
+
+@WebMvcTest(AttachmentsController::class)
+@Import(IntegrationTestConfiguration::class)
+@ActiveProfiles("itest")
+class AttachmentControllerITests(@Autowired override val mockMvc: MockMvc) : ControllerTest {
+
+    @Autowired private lateinit var attachmentService: AttachmentService
+    @Autowired private lateinit var hankeService: HankeService
+    @Autowired private lateinit var permissionService: PermissionService
+
+    @BeforeEach
+    fun clearMocks() {
+        clearAllMocks()
+    }
+
+    @AfterEach
+    fun checkMocks() {
+        confirmVerified(attachmentService)
+    }
+
+    fun send(tunnus: String, file: MockMultipartFile): ResultActions {
+        return mockMvc.perform(
+            MockMvcRequestBuilders.multipart("/liitteet/hanke/${tunnus}")
+                .file(file)
+                .with(SecurityMockMvcRequestPostProcessors.csrf())
+        )
+    }
+
+    @Test
+    fun `Uploading without a session should fail`() {
+        val file = MockMultipartFile("liite", "text.txt", "text/plain", "ABC".toByteArray())
+        send("HAI-123", file).andExpect(status().`is`(401))
+    }
+
+    @Test
+    @WithMockUser(username)
+    fun `Uploading without hankeTunnus should fail`() {
```

Comment:
Isn't this with an unknown hankeTunnus rather than without one?

Reply:
Changed the name to better reflect the test.

Reply:
Was this test removed? I can't find it anymore.